content/reference/custom_interactions.md

```diff
@@ -0,0 +1,221 @@
++++
+title = 'Custom Interactions'
+weight = 6
++++
+
+Buttons, Modals, Select Menus, Ephemeral Messages, and more!
+
+<!--more-->
+
+{{% notice warning %}}
+
+Use of interactions within YAGPDB is an advanced topic. You will need a thorough understanding of YAGPDB's scripting
+language before learning interactions.
+
+We have a comprehensive learning course located [here](https://learn.yagpdb.xyz) for you to get started.
+
+{{% /notice %}}
+
+## The Basics
+
+Interactions within Discord allow server members to use alternative, built-in features to trigger bots to take action
+other than messages or reactions. These features include builtin buttons, dropdown selection menus, or submitting a
+modal (basically a pop-up form). Within custom commands it is possible to both create and customize these new
+interactive features, but respond to them as well, opening up new possibilities for ephemeral message responses, modals,
+and more within YAGPDB custom templating script.
+
+### Definitions
+
+**Interaction** - A user engaging with YAGPDB through one of Discord's builtin features: Clicking a button, Making a
+selection with a select menu, or Submitting a modal.
+
+**Response** - YAGPDB is required to send an initial interaction response after receiving an *interaction*. If it does
+not do this, the user triggering the interaction will see a "This application did not respond" error. The bot cannot
+respond to an interaction more than once. YAGPDB may only respond to an interaction in one of the following ways:
+Sending a message, sending a modal, or updating the message which the *interaction* was triggered on.
```

Comment:
```suggestion
**Response** - YAGPDB is required to respond promptly after receiving an interaction by either sending a message or modal, or by updating the message on which the interaction was triggered. If it does not do this, the user triggering the interaction will see a "This application did not respond" error. The bot cannot respond to an interaction more than once.
```
This reformulation puts the definition of "initial interaction response" up front, which I think is preferable.